### ISSUE TITLE
Question about the hyperparameter

### ISSUE THREAD

**Issue by jxqhhh** [gtolias/mom#1]
I'm really interested in your paper and I want to test the proposed algorithm in the CUB_200_2011 dataset. However, the generated positive pool on my machine only achieves a true positive rate of 28%. I think maybe I just select an improper feature dimension after PCA whitening. I hate to bother you, but I wonder if I could ask you about your selection of the feature dimension after PCA whitening, since it may take just too much time to test on hundreds of possible values to find a proper one.
Thanks！

**Comment by gtolias** [comment]
You can find the set of features Y used for CUB [here](http://cmp.felk.cvut.cz/~toliageo/lala/e/CUB_200_2011_rmac_ms_pcaw512_icp9out_googlenet.mat).

**Comment by jxqhhh** [comment]
Thanks a lot!
I have achieved a very clean positive pool with the set of features you provided. However, the features extracted by myself still failed to work even if I chose the same feature dimension as your features'. I'm wondering whether it's correct to form the initial features of the CUB_200_2011 dataset by using the code [here](https://github.com/gtolias/rmac) to perform R-MAC on the last convolutional feature map of the pre-trained Inception V1 network provided [here](https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v1.py). I would appreciate it very much if you could help me with my problem. 
Thanks!

**Comment by gtolias** [comment]
We have used the pre-trained model provided [here](http://www.vlfeat.org/matconvnet/models/imagenet-googlenet-dag.mat). As mentioned in the paper, descriptors from 3 scales (re-scaling by sqrt(2), 1, 1/sqrt(2)) are aggregated and the final one is whitened.

**Comment by jxqhhh** [comment]
Thank you for your patient and kindly guidance! 
I'm wondering if I correctly understand your response. According to my understanding, I should extract feature descriptors at 3 different scales(l=1,2,3).  Then I should post process it with l2-normalization, re-scaling( the feature vector at the largest scale l = 1 re-scaled by sqrt(2), vectors at scale 2 re-scaled by 1, vectors at scale 3 re-scaled by 1/sqrt(2) ), PCA-whitening and l2-normalization.
I'm wondering if my understanding is correct. Could you help me with that? Thanks a lot!

**Comment by jxqhhh** [comment]
Thank you for your patient and kindly guidance.
I'm still a little confused about how to get R-MAC descriptors. I would really appreciate it if you could help me with the following problems.
1. When should I re-scale descriptors from 3 different scales? I don't know if the following order is correct: l2-normalization--->re-scaling descriptors--->learning PCA--->perform PCA-whiten--->summing all regional feature vectors to get a single image vector--->l2-normalization. 
2. I don't know which one should be re-scaled by sqrt(2).Should I re-scale descriptors form scale l=1 by sqrt(2) or re-scale descriptors form scale l=3 by sqrt(2) ?
Thanks a lot！

**Comment by jxqhhh** [comment]
Thanks a lot!
I think maybe my code have some serious problems and I have contacted you via email. Thank so much for your patient and kindly guidance!

**Comment by gtolias** [comment]
[This](https://github.com/gtolias/mom/blob/master/extract_CUB_descriptors_pretrained.m) newly added script should clarify the extraction process for vector set Y (in the paper) on CUB.

**Comment by jxqhhh** [comment]
Thank you so much for your patient help!
